tests/js/realm-tests.js

```diff
@@ -970,8 +968,7 @@ module.exports = {
         const realm2 = new Realm({schema: [schemas.StringOnly], shouldCompactOnLaunch: shouldCompact});
         TestCase.assertTrue(wasCalled);
         TestCase.assertEqual(realm2.objects('StringOnlyObject').length, count + 2);
-        var afterSize = fs.statSync(realm2.path).size;
-        TestCase.assertTrue(beforeSize >= afterSize);
+        // we don't check if the file is smaller as we assume that Object Store does it
```

Comment:
Yeah, it's just a nitpick - we don't have a test that validates that the result (`true`) that we set in the js callback is what's getting delivered to the OS callback. I'm sure it is, but would be nice to test it :) In any case, if it's hard, I guess we can skip it and assume it works.